### PR TITLE
UCT/IB/MLX5/DC: dci_channel_index is not limited by num_dci_channels

### PR DESCRIPTION
**Pull request by roiedanino**
## What
Removed limitation of `dci_channel_index` by `num_dci_channels`

## Why ?
Improvement in performance, utilization of all 256 DCI channels

### CODE REVIEW

File: src/uct/ib/mlx5/dc/dc_mlx5_ep.h [hunk 1]
```diff
@@ -368,8 +368,7 @@ uct_dc_mlx5_ep_basic_init(uct_dc_mlx5_iface_t *iface, uct_dc_mlx5_ep_t *ep)
         }
         ep->dci               = 0;
         dci                   = uct_dc_mlx5_iface_dci(iface, ep->dci);
-        ep->dci_channel_index = dci->next_channel_index++ %
-                                iface->tx.num_dci_channels;
+        ep->dci_channel_index = dci->next_channel_index++;
```
Comment (gleon99) [inline src/uct/ib/mlx5/dc/dc_mlx5_ep.h:371]: Can't it grow infinitely?

Comment (roiedanino) [reply src/uct/ib/mlx5/dc/dc_mlx5_ep.h:371]: dci_channel_index is uint8_t so it will go back to 0 after 255

Comment (brminich) [reply src/uct/ib/mlx5/dc/dc_mlx5_ep.h:371]: how is it handled in HW? Do they do module by channel num themselves?

**Comment by roiedanino** [comment]
> is there a specific test case that improved because of this change?

osu_all2all (when setting policy to hw_dcs)

**Pull request merged**